### PR TITLE
feat: `docs` cmd to generate `zetacored` CLI docs

### PR DESCRIPTION
**Pull request by fadeev**
* `zetacored docs`: a command to generate CLI docs
* `gen-docs-zetacored.sh`:
  * generate docs,
  * reformat placeholders so that Docusaurus doesn't think they're links
  * rename heading to remove `zetacored`, so that they're rendered nicely in the sidebar
* removed `flags.LineBreak,` in `tx`, because it generated a link to a non-existent file
* added `AppName` to config with a value of `zetacored`

Screenshot 2023-08-14 at 11 53 02

### CODE REVIEW

**Comment by fadeev** [comment]
If this works, we can do the same for `zetacliend` docs.

File: cmd/zetacore_utils/main.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 	sdkmath "cosmossdk.io/math"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/zeta-chain/zetacore/cmd/zetacored/config"
+	zetacoredconfig "github.com/zeta-chain/zetacore/cmd/zetacored/config"
```
Comment (lumtis) [inline cmd/zetacore_utils/main.go:15]: Imported twice

File: cmd/zetacored/docs.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		return err
+	}
+	return nil
+}
+
+func docsCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "docs [path]",
+		Short: "Generate markdown documentation for zetacored",
+		RunE:  docsCmd,
```
Comment (lumtis) [inline cmd/zetacored/docs.go:42]:
```suggestion
		RunE:  docsCmd,
```
```suggestion
		RunE:  docsCmd,
		Args:  cobra.MaximumNArgs(1),
```

**Comment by fadeev** [comment]
The "Generated Files" is failing because of paths:

```
-      --home string              The application home directory (default "/Users/fadeev/.zetacored")
+      --home string              The application home directory (default "/home/runner/.zetacored")
```

**Comment by fadeev** [comment]
GH workflow is generating docs with `[appd]` instead of `zetacored` 😭

**Review by brewmaster012 — APPROVED** [review]
Cool!!

**Comment by fadeev** [comment]
Can't merge, because I'm getting a billion linter errors unrelated to this PR:

```
make lint                                                                 
all modules verified
rpc/backend/node_info.go:252:2: unused-parameter: parameter 'language' seems to be unused, consider removing or renaming it as _ (revive)
        language keyring.Language,
        ^
rpc/namespaces/ethereum/eth/filters/filters.go:110:23: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (f *Filter) Logs(ctx context.Context, logLimit int, blockLimit int64) ([]*ethtypes.Log, error) {
                      ^
rpc/namespaces/ethereum/eth/api.go:356:49: unused-parameter: parameter 'hash' seems to be unused, consider removing or renaming it as _ (revive)
func (e *PublicAPI) GetUncleByBlockHashAndIndex(hash common.Hash, idx hexutil.Uint) map[string]interface{} {
                                                ^
rpc/namespaces/ethereum/eth/api.go:361:51: unused-parameter: parameter 'number' seems to be unused, consider removing or renaming it as _ (revive)
func (e *PublicAPI) GetUncleByBlockNumberAndIndex(number, idx hexutil.Uint) map[string]interface{} {
                                                  ^
rpc/namespaces/ethereum/eth/api.go:366:46: unused-parameter: parameter 'hash' seems to be unused, consider removing or renaming it as _ (revive)
func (e *PublicAPI) GetUncleCountByBlockHash(hash common.Hash) hexutil.Uint {
                                             ^
rpc/namespaces/ethereum/eth/api.go:371:48: unused-parameter: parameter 'blockNum' seems to be unused, consider removing or renaming it as _ (revive)
func (e *PublicAPI) GetUncleCountByBlockNumber(blockNum rpctypes.BlockNumber) hexutil.Uint {
                                               ^
rpc/namespaces/ethereum/miner/unsupported.go:34:26: unused-parameter: parameter 'extra' seems to be unused, consider removing or renaming it as _ (revive)
func (api *API) SetExtra(extra string) (bool, error) {
                         ^
rpc/namespaces/ethereum/miner/unsupported.go:42:29: unused-parameter: parameter 'gasLimit' seems to be unused, consider removing or renaming it as _ (revive)
func (api *API) SetGasLimit(gasLimit hexutil.Uint64) bool {
                            ^
rpc/namespaces/ethereum/miner/unsupported.go:54:23: unused-parameter: parameter 'threads' seems to be unused, consider removing or renaming it as _ (revive)
func (api *API) Start(threads *int) error {
                      ^
cmd/zetacore_utils/main.go:79:9: G204: Subprocess launched with a potential tainted input or cmd arguments (gosec)
        cmd := exec.Command(config.AppName, args...)
               ^
common/address.go:29:42: empty-block: this block is empty, you can remove it (revive)
        } else if IsBitcoinChain(chain.ChainId) {

        }
x/emissions/types/codec.go:9:20: unused-parameter: parameter 'cdc' seems to be unused, consider removing or renaming it as _ (revive)
func RegisterCodec(cdc *codec.LegacyAmino) {
                   ^
x/emissions/client/cli/query.go:17:18: unused-parameter: parameter 'queryRoute' seems to be unused, consider removing or renaming it as _ (revive)
func GetQueryCmd(queryRoute string) *cobra.Command {
                 ^
x/emissions/keeper/params.go:9:27: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) GetParams(ctx sdk.Context) types.Params {
                          ^
x/emissions/keeper/grpc_query_list_balances.go:11:35: unused-parameter: parameter 'ctx' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) ListPoolAddresses(ctx context.Context, req *types.QueryListPoolAddressesRequest) (*types.QueryListPoolAddressesResponse, error) {
                                  ^
x/observer/client/cli/query.go:17:18: unused-parameter: parameter 'queryRoute' seems to be unused, consider removing or renaming it as _ (revive)
func GetQueryCmd(queryRoute string) *cobra.Command {
                 ^
x/emissions/module.go:65:60: unused-parameter: parameter 'config' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
                                                           ^
x/emissions/module.go:74:42: unused-parameter: parameter 'clientCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Router) {
                                         ^
x/emissions/module.go:78:49: unused-parameter: parameter 'clientCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
                                                ^
x/emissions/module.go:129:42: unused-parameter: parameter 'legacyQuerierCdc' seems to be unused, consider removing or renaming it as _ (revive)
func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
                                         ^
x/fungible/client/cli/query.go:17:18: unused-parameter: parameter 'queryRoute' seems to be unused, consider removing or renaming it as _ (revive)
func GetQueryCmd(queryRoute string) *cobra.Command {
                 ^
x/observer/keeper/observer_mapper.go:89:55: unused-parameter: parameter 'msg' seems to be unused, consider removing or renaming it as _ (revive)
func (k msgServer) AddObserver(goCtx context.Context, msg *types.MsgAddObserver) (*types.MsgAddObserverResponse, error) {
                                                      ^
rpc/websockets.go:685:40: unused-parameter: parameter 'wsConn' seems to be unused, consider removing or renaming it as _ (revive)
func (api *pubSubAPI) subscribeSyncing(wsConn *wsConn, subID rpc.ID) (pubsub.UnsubscribeFunc, error) {
                                       ^
x/fungible/keeper/begin_blocker_deploy_system_contracts.go:10:47: unused-parameter: parameter 'goCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) BlockOneDeploySystemContracts(goCtx context.Context) error {
                                              ^
x/fungible/keeper/begin_blocker_deploy_system_contracts.go:13:49: unused-parameter: parameter 'goCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) TestUpdateSystemContractAddress(goCtx context.Context) error {
                                                ^
x/fungible/keeper/begin_blocker_deploy_system_contracts.go:16:44: unused-parameter: parameter 'goCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) TestUpdateZRC20WithdrawFee(goCtx context.Context) error {
                                           ^
app/ante/handler_options.go:173:18: unused-parameter: parameter 'simulate' seems to be unused, consider removing or renaming it as _ (revive)
func SetGasMeter(simulate bool, ctx sdk.Context, gasLimit uint64) sdk.Context {
                 ^
x/fungible/module_simulation.go:39:40: unused-parameter: parameter 'simState' seems to be unused, consider removing or renaming it as _ (revive)
func (am AppModule) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
                                       ^
x/fungible/module.go:66:60: unused-parameter: parameter 'config' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
                                                           ^
x/fungible/module.go:75:42: unused-parameter: parameter 'clientCtx' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Router) {
                                         ^
x/fungible/module.go:137:42: unused-parameter: parameter 'legacyQuerierCdc' seems to be unused, consider removing or renaming it as _ (revive)
func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
                                         ^
x/observer/module.go:66:60: unused-parameter: parameter 'config' seems to be unused, consider removing or renaming it as _ (revive)
func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
                                                           ^
x/observer/module.go:137:42: unused-parameter: parameter 'legacyQuerierCdc' seems to be unused, consider removing or renaming it as _ (revive)
func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sdk.Querier {
                                         ^
x/crosschain/keeper/zeta_conversion_rate.go:44:29: unused-parameter: parameter 'context' seems to be unused, consider removing or renaming it as _ (revive)
func (k Keeper) ProtocolFee(context context.Context, req *types.QueryMessagePassingProtocolFeeRequest) (*types.QueryMessagePassingProtocolFeeResponse, error) {
                            ^
x/crosschain/module_simulation.go:39:40: unused-parameter: parameter 'simState' seems to be unused, consider removing or renaming it as _ (revive)
func (am AppModule) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
                                       ^
zetaclient/zetabridge.go:119:60: unused-parameter: parameter 'keyType' seems to be unused, consider removing or renaming it as _ (revive)
func (b *ZetaCoreBridge) GetAccountNumberAndSequenceNumber(keyType common.KeyType) (uint64, uint64, error) {
                                                           ^
zetaclient/evm_signer.go:426:57: unused-parameter: parameter 'recipient' seems to be unused, consider removing or renaming it as _ (revive)
func (signer *EVMSigner) SignWhitelistTx(action string, recipient ethcommon.Address, asset ethcommon.Address, gasLimit uint64, nonce uint64, gasPrice *big.Int, height uint64) (*ethtypes.Transaction, error) {
                                                        ^
make: *** [lint] Error 1
```

**Comment by lumtis** [comment]
It looks like Linter is running `gocritic` https://github.com/go-critic/go-critic while we don't have it in the CI and it is not triggered in other PR, I don't see why

**Comment by fadeev** [comment]
Even with https://github.com/zeta-chain/node/pull/997 the linter still fails.

File: cmd/zetacore_utils/main.go [hunk 3]
```diff
@@ -85,7 +85,7 @@ func main() {
 	args = append(args, []string{distributionList[0].TokensDistributed.String(), "--keyring-backend", "test", "--chain-id", chainID, "--yes",
 		"--broadcast-mode", broadcastMode, "--gas=auto", "--gas-adjustment=2", "--gas-prices=0.001azeta", "--node", node}...)
 
-	cmd := exec.Command("zetacored", args...)
+	cmd := exec.Command(config.AppName, args...)
```
Comment (lumtis) [inline cmd/zetacore_utils/main.go:88]: Using a variable for the command name is what trigger the gosec issue.
I think we should follow the linter and continue to use the hardcoded `zetacored` for the command name

File: cmd/zetacore_utils/main.go [hunk 4]
```diff
@@ -54,7 +54,7 @@ func main() {
 
 	distributionList := make([]TokenDistribution, len(addresses))
 	for i, address := range addresses {
-		cmd := exec.Command("zetacored", "q", "bank", "balances", address, "--output", "json", "--denom", "azeta", "--node", node) // #nosec G204
+		cmd := exec.Command(config.AppName, "q", "bank", "balances", address, "--output", "json", "--denom", "azeta", "--node", node) // #nosec G204
```
Comment (lumtis) [inline cmd/zetacore_utils/main.go:57]: same comment

File: cmd/zetacore_utils/main.go [hunk 5]
```diff
@@ -97,7 +97,7 @@ func main() {
 	time.Sleep(7 * time.Second)
 
 	for i, address := range addresses {
-		cmd := exec.Command("zetacored", "q", "bank", "balances", address, "--output", "json", "--denom", "azeta", "--node", node) // #nosec G204
+		cmd := exec.Command(config.AppName, "q", "bank", "balances", address, "--output", "json", "--denom", "azeta", "--node", node) // #nosec G204
```
Comment (lumtis) [inline cmd/zetacore_utils/main.go:100]: Same comment

**Review by lumtis — COMMENTED** [review]
I think we can make the PR open again?

**Review by lumtis — APPROVED** [review]

**Review by CryptoFewka — APPROVED** [review]
The workflow change looks good/safe.

**Pull request merged**